phoenix-core/src/main/java/org/apache/phoenix/util/UpgradeUtil.java

```diff
@@ -1279,4 +1304,129 @@ public static boolean truncateStats(HTableInterface metaTable, HTableInterface s
         }
         return false;
     }
+
+    public static void mapTableToNamespace(HBaseAdmin admin, HTableInterface metatable, String srcTableName,
+            String destTableName, ReadOnlyProps props, Long ts, String phoenixTableName, PTableType pTableType)
+                    throws SnapshotCreationException, IllegalArgumentException, IOException, InterruptedException,
+                    SQLException {
+        srcTableName = SchemaUtil.normalizeIdentifier(srcTableName);
+        if (!SchemaUtil.isNamespaceMappingEnabled(
+                SchemaUtil.isSystemTable(srcTableName.getBytes()) ? PTableType.SYSTEM : null,
+                props)) { throw new IllegalArgumentException(SchemaUtil.isSystemTable(srcTableName.getBytes())
+                        ? "For system table " + QueryServices.IS_SYSTEM_TABLE_MAPPED_TO_NAMESPACE
+                                + " also needs to be enabled along with " + QueryServices.IS_NAMESPACE_MAPPING_ENABLED
+                        : QueryServices.IS_NAMESPACE_MAPPING_ENABLED + " is not enabled"); }
+
+        if (PTableType.TABLE.equals(pTableType) || PTableType.INDEX.equals(pTableType)) {
+            admin.snapshot(srcTableName, srcTableName);
+            admin.cloneSnapshot(srcTableName.getBytes(), destTableName.getBytes());
+            admin.disableTable(srcTableName);
+            admin.deleteTable(srcTableName);
+        }
+        if (phoenixTableName == null) {
```

Comment:
phoenixTableName and physical table name are different in case of indexes .. so it is required there .. but now in my latest commit , I have removed the null check and pass the phoenixTableName always.